src/shogun/base/SGObject.h

```diff
@@ -124,6 +124,16 @@ class CSGObject
 	/** destructor */
 	virtual ~CSGObject();
 
+	void saveJSON(char* filename) const;
```

Comment:
great! so in the same manner, lets add support for all the default cereal output formats: xml, binary.